jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpGenerator.java

```diff
@@ -627,23 +628,36 @@ else if (contentLength != field.getLongValue())
 
                         case CONNECTION:
                         {
-                            boolean keepAlive = field.contains(HttpHeaderValue.KEEP_ALIVE.asString());
-                            if (keepAlive && _info.getHttpVersion() == HttpVersion.HTTP_1_0 && _persistent == null)
+                            boolean hasValue = StringUtil.isNotBlank(field.getValue());
+                            boolean hasKeepAlive = field.contains(HttpHeaderValue.KEEP_ALIVE.asString());
+                            boolean hasClose = field.contains(HttpHeaderValue.CLOSE.asString());
```

Comment:
Whilst this style is easily readable, it contains a lot of redundant code/tests.  If `hasValue==false`, then you can break straight away and avoid subsequent work.

If it `hasKeepAlive` is true, then in your code there is no need to lookup `hasClose` (but see below).